ddtrace/internal/runtime/__init__.py

```diff
@@ -14,39 +15,15 @@ def _generate_runtime_id():
 _RUNTIME_ID = _generate_runtime_id()
 
 
-if hasattr(os, "register_at_fork"):
-
-    def _set_runtime_id():
-        global _RUNTIME_ID
-        _RUNTIME_ID = _generate_runtime_id()
-
-    os.register_at_fork(after_in_child=_set_runtime_id)
-
-    def get_runtime_id():
-        return _RUNTIME_ID
-
-
-else:
-    # Non-POSIX systems or Python < 3.7
-    import threading
-
-    _RUNTIME_PID = os.getpid()
-
-    def _set_runtime_id():
-        global _RUNTIME_ID, _RUNTIME_PID
-        _RUNTIME_ID = _generate_runtime_id()
-        _RUNTIME_PID = os.getpid()
-
-    _RUNTIME_LOCK = threading.Lock()
-
-    def get_runtime_id():
-        with _RUNTIME_LOCK:
-            pid = os.getpid()
-            if pid != _RUNTIME_PID:
-                _set_runtime_id()
-            return _RUNTIME_ID
+@forksafe.register
```

Comment:
I think you'd want to register within the function to allow subsequent forks to run the same function

Reply:
Ah yes I was thinking about that 🤔... Wouldn't the same hooks be "inherited" by the child process? I'll confirm with some test cases if we don't already have them.

Reply:
Hmm, I was convinced that hooks were discarded once executed, but the code suggests otherwise :\. So this should work.

Reply:
Yeah looks like we're good: #2561